homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -236,6 +245,149 @@ class EnvoyConsumptionSensorEntityDescription(
+        device_class=SensorDeviceClass.FREQUENCY,
+        suggested_unit_of_measurement=UnitOfFrequency.HERTZ,
+        suggested_display_precision=1,
+        entity_registry_enabled_default=False,
+        value_fn=lambda ct: ct.frequency,
+        on_phase=None,
+    ),
+    EnvoyCTSensorEntityDescription(
+        key="voltage",
+        translation_key="net_ct_voltage",
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        suggested_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        suggested_display_precision=1,
+        entity_registry_enabled_default=False,
+        value_fn=lambda ct: ct.voltage,
+        on_phase=None,
+    ),
+    EnvoyCTSensorEntityDescription(
+        key="net_consumption_ct_metering_status",
+        translation_key="net_ct_metering_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=list(CtMeterStatus),
+        entity_registry_enabled_default=False,
+        value_fn=lambda ct: ct.metering_status,
+        on_phase=None,
+    ),
+    EnvoyCTSensorEntityDescription(
+        key="net_consumption_ct_status_flags",
+        translation_key="net_ct_status_flags",
+        state_class=None,
+        entity_registry_enabled_default=False,
+        value_fn=lambda ct: 0 if ct.status_flags is None else len(ct.status_flags),
+        on_phase=None,
+    ),
+)
+
+
+CT_NET_CONSUMPTION_PHASE_SENSORS = {
+    (on_phase := str(PhaseNames(PHASENAMES[phase]))): [
+        replace(
+            sensor,
+            key=f"{sensor.key}_l{phase + 1}",
+            translation_key=f"{sensor.translation_key}_phase",
+            on_phase=on_phase,
+            translation_placeholders={"phase_name": f"l{phase + 1}"},
+        )
+        for sensor in list(CT_NET_CONSUMPTION_SENSORS)
+    ]
+    for phase in range(0, 3)
+}
+
+CT_PRODUCTION_SENSORS = (
+    EnvoyCTSensorEntityDescription(
+        key="production_ct_metering_status",
+        translation_key="production_ct_metering_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=list(CtMeterStatus),
+        entity_registry_enabled_default=False,
+        value_fn=lambda ct: ct.metering_status,
+        on_phase=None,
+    ),
+    EnvoyCTSensorEntityDescription(
+        key="production_ct_status_flags",
+        translation_key="production_ct_status_flags",
+        state_class=None,
+        entity_registry_enabled_default=False,
+        value_fn=lambda ct: 0 if ct.status_flags is None else len(ct.status_flags),
+        on_phase=None,
+    ),
+)
+
+CT_PRODUCTION_PHASE_SENSORS = {
+    (on_phase := str(PhaseNames(PHASENAMES[phase]))): [
+        replace(
+            sensor,
+            key=f"{sensor.key}_l{phase + 1}",
+            translation_key=f"{sensor.translation_key}_phase",
+            on_phase=on_phase,
+            translation_placeholders={"phase_name": f"l{phase + 1}"},
```

Comment:
I think the additional phases should be disabled by default entities since they will take up a lot of space in the database and most users will only want the aggregate anyways

Reply:
Done. We can do same for the existing Muliphase ones. Won't impact existing users and let new users choose (the reverse way from now).